test/error/bad_double_buffer.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Halide.h"
+#include <stdio.h>
+
+using namespace Halide;
+
+int main(int argc, char **argv) {
+    Func f("f"), g("g"), h("h");
+    Var x("x"), y("y");
+
+    f(x) = x;
+    g(x) = f(x);
+    h(x, y) = g(x);
+
+    g.compute_at(h, y);
+
+    // ring_buffer() requires an explicit hoist_storage().
```

Comment:
This seems to suggest the implicit parameter I was wondering above. I think this needs to go in both the tutorial and the `Func::ring_buffer()` documentation.

Reply:
Added more comments about it in both places.